Test_Sender.cpp

```diff
@@ -0,0 +1,29 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "catch.hpp"
+#include "BMS_Sender.h"
+
+
+TEST_CASE("Case To Check the Input Data from file") {
+  InputType Source= FileInput;
+  REQUIRE(Read_Input_Data(Source) == Success);
+}
+
+TEST_CASE("Case To Check the Input Data from Random generation") {
+  InputType Source= RandomGeneration;
+  REQUIRE(Read_Input_Data(Source) == Success);
+}
+
+TEST_CASE("Case to print to console") {
+  InputType Source= FileInput;
+  Targettype Target= ToConsole;
+  REQUIRE(Read_Input_Data(Source) == Success);
+  REQUIRE(OutputtoTarget(Target) == Success);
+}
+

```

Comment:
These tests will pass even if you change the output format and make this program incompatible with the receiver. Focus on testing your functionality - the most crucial is the output format.
In the absence of a test for the output format, any output-mistakes by future maintainers will only be caught only in integration with the receiver.